sw/host/tests/chip/mem/src/main.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use clap::Parser;
+use std::collections::HashMap;
+use std::fs;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use object::{Object, ObjectSymbol};
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::mem::{MemRead32Req, MemReadReq, MemWrite32Req, MemWriteReq};
+use opentitanlib::uart::console::UartConsole;
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[command(flatten)]
+    init: InitializeTest,
+
+    /// Console receive timeout.
+    #[arg(long, value_parser = humantime::parse_duration, default_value = "600s")]
+    timeout: Duration,
+
+    /// Path to the firmware's ELF file, for querying symbol addresses.
+    #[arg(long, default_value = None)]
+    firmware_elf: Option<PathBuf>,
```

Comment:
If you want this arg to be required there are a couple of options:

1. If you make this a positional arg instead of an commandline flag, clap will require it to be provided.
   ```
   #[arg(value_name = "FIRMWARE_ELF")]
   firmware_elf: PathBuf,
   ```

2. If you want this to be a required flag, you can say `required=true`:
   ```
   #[arg(long, required=true)]
   firmware_elf: PathBuf,
   ```

Both of these should do away with the `let Some(ref blah) = ...` business below.